services/api/service.go

```diff
@@ -449,6 +465,45 @@ func (api *RelayAPI) startValidatorRegistrationDBProcessor() {
 	}
 }
 
+func (api *RelayAPI) processPayloadAttributes(payloadAttributes beaconclient.PayloadAttributesData) {
+	// only for builder-api and if using see subscriptions instead of querying for payload attributes
+	if api.opts.BlockBuilderAPI && api.ffDisableQueryPayloadAttributes {
+		apiHeadSlot := api.headSlot.Load()
+		proposalSlot := payloadAttributes.Data.ProposalSlot
+		if proposalSlot <= apiHeadSlot {
+			return
+		}
+		log := api.log.WithField("proposalSlot", proposalSlot)
+
+		log.Info("updating payload attributes")
```

Comment:
this function will be called multiple times for each payload attributes -- namely once per CL client. therefore this function should somehow guard against it.

question: do we need to support multiple parents? or do we just always use the latest received one?